### DISCUSSION TITLE
[bedrock] Error: Parameter validation failed: Invalid type for parameter system, value: type: <class 'str'>, valid types: <class 'list'>, <class 'tuple'>

### DISCUSSION THREAD

**Discussion by zytbeyond** [langgenius/dify#5828] [Help]
In Dify SaaS v0.6.12-fix1, system prompts appear to be non-functional with AWS Bedrock Claude3. You can reproduce this issue by initiating any chat template and selecting AWS Bedrock Claude3 as the LLM.
![image](https://github.com/langgenius/dify/assets/58652299/5817a84e-8116-468b-9d8b-8b3677668da4)

**Comment by kimnamu** [comment]
Thanks for the clear report, and for pasting the exact error text: it pins the cause to a three-day window on `main`. The error came from PR #5768 (merged 2024-06-30, one day after the 0.6.12-fix1 tag), which moved Bedrock Claude 3 to the Converse API but still passed the system prompt as a string. PR #5889 (2024-07-03) changed it to the `[{"text": ...}]` list the API expects, and that shipped in 0.6.13.

| Build | what `_convert_converse_prompt_messages` returns | `converse(system=...)` |
|---|---|---|
| 0.6.12-fix1 tag | not used (Anthropic SDK path, where a string `system` is valid) | not called |
| `main` between #5768 and #5889 (2024-06-30 to 07-03) | `str` | `ParamValidationError`, the text in your screenshot |
| 0.6.13 and later | `list` of `{"text": ...}` | accepted |

Stand-in run of both versions of that function against boto3's client-side validator, with dummy credentials so the request is checked before it is signed:

```text
boto3 1.40.76
--- fdfbbde10d (#5768): system='You are a helpful assistant.'
ParamValidationError: Parameter validation failed: Invalid type for parameter system, value: You are a helpful assistant., type: <class 'str'>, valid types: <class 'list'>, <class 'tuple'>
--- 0.6.13 (#5889): system=[{'text': 'You are a helpful assistant.'}]
ClientError: ResourceNotFoundException - This model version has reached the end of its life. Please refer to the AWS documentation for more details.
```

The second run gets past validation and reaches AWS (the 2024 model ID is retired today, hence that message). A self-hosted install only sees this error when built from `main` in that window; tagged releases, 0.6.12-fix1 included, never had it, and 0.6.13 or later fixes it.

<details>
<summary>Code paths I checked</summary>

- #5768 (`fdfbbde10d`) `api/core/model_runtime/model_providers/bedrock/llm/llm.py` L314-324: `system = []` is initialised, then `system = message.content` overwrites it with the stripped string on the first `SystemPromptMessage`. L178 and L187 pass it straight to `converse_stream` / `converse`, whose `system` parameter is a list of `SystemContentBlock`.
- #5889 (`aecdfa2d5c`) L342-346: `system.append({"text": message.content})`, and L182-183 only add `system` to the call when the list is non-empty.
- No tagged release carries the broken version: `git tag --contains fdfbbde10d` starts at 0.6.13, which already includes #5889. Only `main` builds between the two merges hit it, and the error text only exists on that path, so the "v0.6.12-fix1" label in the footer pointed at the last tag rather than the deployed commit.
- The Bedrock runtime now lives in `langgenius/dify-official-plugins`, `models/bedrock/models/llm/llm.py` L1131, and keeps the list form.
</details>
